### PR TITLE
Add counting to relevant requests endpoint

### PR DESCRIPTION
**Pull request by loldot**
- [ ] New feature
- [x] Bug fix
- [ ] High impact

**Description of work:**
Add `$count` parameter to unassigned requests for department endpoint

**Testing:**
- [x] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

**Checklist:**
- [x] Considered automated tests
- [x] Considered updating specification / documentation
- [x] Considered work items 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Domain/Queries/GetResourceAllocationRequests.cs [hunk 1]
```diff
@@ -185,9 +191,10 @@ public async Task<QueryRangedList<QueryResourceAllocationRequest>> Handle(GetRes
                 if (!countOnly)
                 {
                     await AddWorkFlows(pagedQuery);
-                    await AddOrgPositions(pagedQuery, request.Expands);
                     await AddProposedPersons(pagedQuery);
                 }
+                
+                await AddOrgPositions(pagedQuery, request.Expands);
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Domain/Queries/GetResourceAllocationRequests.cs:197]: Is there a reason we want to resolve the positions when only the count is relevant?

Comment (loldot) [reply src/backend/api/Fusion.Resources.Domain/Queries/GetResourceAllocationRequests.cs:197]: Yes, we need the information on base positions when we try to filter out the requests that are relevant for that department.

Comment (loldot) [reply src/backend/api/Fusion.Resources.Domain/Queries/GetResourceAllocationRequests.cs:197]: nvm, I changed another part of the implementation forgot to change this back. Will update

**Review by HansDahle — APPROVED** [review]
Nice, did not think to follow the count pattern here.

**Pull request merged**